pkg/httputil/proxy/proxy.go

```diff
@@ -154,8 +155,11 @@ func AwareDialContextTLS(ctx context.Context, address string, tlsClientConf *tls
 		tlsClientConf = tlsClientConf.Clone()
 		tlsClientConf.ServerName = host
 	}
+	ctxHandshake, cancel := context.WithTimeout(ctx, tlsHandshakeTimeout)
+	defer cancel()
 	tlsConn := tls.Client(conn, tlsClientConf)
-	if err := tlsConn.Handshake(); err != nil {
+	// The handshake must be done with a timeout to avoid infinite blocking of Sensor sync.
+	if err := tlsConn.HandshakeContext(ctxHandshake); err != nil {
```

Comment:
hardcoded 2s timeout feels arbitrary, any reason for not re-using `ctx` function param?

Reply:
Yes, 2s are arbitrary here, but that was the intention of the original author of `CheckTLS`, cause the parent context has timeout of 2s and it was meant to be used for the entire check, not only for the handshake.

I am using here a separate context, as the `defaultDialer.DialContext` may replace the original context with its own one.  Due to that, `DialContext` may take longer than 2 seconds and that would make the parent context either to: 
- have already expired - which is good as it would make the `ctxHandshake` also expired  and would not block, 
- have changed timeout by  - which is bad as we lost control over the timeout value.

Reply:
If `defaultDialer.DialContext` may replace the context that's passed to it, could we isolate the context by passing a copy of the parent context?

Reply:
I did some tests and the context is being changed locally for `DialContext`. Outside of it, the context deadline should not be affected.